addons/hr_expense/models/hr_expense.py

```diff
@@ -1031,6 +1034,17 @@ def create(self, vals):
         sheet.activity_update()
         return sheet
 
+    def write(self, vals):
+        if 'state' in vals:
+            # Avoid user with write access on expense sheet in draft state to bypass the validation process
+            if not self.user_has_groups('hr_expense.group_hr_expense_manager') and self.state == 'draft' and vals['state'] != 'submit':
+                raise UserError(_("You don't have the rights to bypass the validation process of this expense report."))
+            if vals['state'] == 'approve':
+                self._check_can_approve()
+            if vals['state'] == 'cancel':
```

Comment:
```suggestion
            elif vals['state'] == 'cancel':
```
Else resetting to draft would test it thrice